trillian/ctfe/instance.go

```diff
@@ -196,16 +199,22 @@ type InstanceOptions struct {
 
 	// CertificateQuotaUser returns a string represeing the passed in
 	// intermediate certificate. This string will be user as a User quota key for
-	// the cert.  Quota will be requested for each intermediate in an
+	// the cert. Quota will be requested for each intermediate in an
 	// add-[pre]-chain request so as to allow individual issers to be rate
-	// limited.  If unset, no quota will be requested for intermediate
+	// limited. If unset, no quota will be requested for intermediate
 	// certificates.
 	CertificateQuotaUser func(*x509.Certificate) string
 }
 
-// SetUpInstance sets up a log instance that uses the specified client to communicate
-// with the Trillian RPC back end.
-func SetUpInstance(ctx context.Context, client trillian.TrillianLogClient, cfg *configpb.LogConfig, opts InstanceOptions) (*PathHandlers, error) {
+// SetUpInstance sets up a log instance that uses the specified client to
+// communicate with the Trillian RPC back end.
+func SetUpInstance(
+	ctx context.Context,
+	client trillian.TrillianLogClient,
+	cfg *configpb.LogConfig,
+	opts InstanceOptions,
+	mfact MirrorSTHFactory,
```

Comment:
Should this be part of InstanceOptions?

Reply:
Hmmmm. Maybe not? Not sure.

Reply:
Yeah, there are multiple imperfect ways to do this... Maybe we could even avoid the factory.